web/react/components/filtered_user_list.jsx

```diff
@@ -37,10 +44,47 @@ class FilteredUserList extends React.Component {
         }
     }
 
-    handleFilterChange(e) {
-        this.setState({
-            filter: e.target.value
-        });
+    searchProfiles() {
```

Comment:
Yeah, most of the problem is that it's synchronous. If it's changed to just an async call, you could move it out to a listener so the UserList wouldn't have to care where its data came from. We'd still be duplicating some code doing that, but the duplicated code would just be the parents calling
```jsx
onFilterChanged(filter) {
    AsyncClient.searchProfiles(filter);
}
```

Also, right on the refs vs e.target thing. I realized why you were doing it when I was looking at the code in the first place and then proceeded to immediately forget that as I started writing my comment.